openmp/runtime/unittests/CMakeLists.txt

```
@@ -0,0 +1,42 @@
+add_custom_target(OpenMPUnitTests)
+set_target_properties(OpenMPUnitTests PROPERTIES FOLDER "OpenMP/Tests")
+
+if(NOT TARGET llvm_gtest)
+  message(WARNING "OpenMP unittests disabled due to GTest being unavailable; "
+                  "Try LLVM_INSTALL_GTEST=ON for the LLVM build")
+  return()
+endif()
+if(NOT TARGET omp_testing)
+  message(WARNING "OpenMP unittests disabled due to omp_testing library being unavailable")
+  return()
+endif()
+
+function(add_openmp_unittest test_filename)
+  add_unittest(OpenMPUnitTests ${test_filename} ${ARGN})
+
+  # Link against the testing library which exports all symbols.
+  target_link_libraries(${test_filename} PRIVATE omp_testing)
+  add_dependencies(${test_filename} omp_testing)
+
+  target_include_directories(${test_filename} PRIVATE
+    ${LIBOMP_INCLUDE_DIR}
+    ${LIBOMP_SRC_DIR}
+  )
```

```suggestion
  target_include_directories(omp_testing PUBLIC
    ${LIBOMP_INCLUDE_DIR}
    ${LIBOMP_SRC_DIR}
  )
```
target_* build options are transitively added to all link-dependencies if they are PUBLIC. If added to `omp_testing`, everything that links to it receive its PUBLIC (and INTERFACE) build options.